src/UsgsAstroLsSensorModel.cpp

```diff
@@ -2634,9 +2635,27 @@ std::string UsgsAstroLsSensorModel::constructStateFromIsd(const std::string imag
   MESSAGE_LOG("m_referencePointXyz: {} ",
                         state["m_referencePointXyz"].dump())
 
-  // sun_position and velocity are required for getIlluminationDirection
-  state["m_sunPosition"]= getSunPositions(isd, parsingWarnings);
-  state["m_sunVelocity"]= getSunVelocities(isd, parsingWarnings);
+  // Sun position and velocity are required for getIlluminationDirection
+  ale::States sunState = ale::getSunPosition(jsonIsd);
+  std::vector<ale::State> sunStates = sunState.getStates();
+  std::vector<double> ephemTime = sunState.getTimes();
+  ale::Orientations j2000_to_target = ale::getBodyRotation(jsonIsd);
+  ale::State rotatedSunState;
+  std::vector<double> sunPositions = {};
+  std::vector<double> sunVelocities = {};
+
+  for (int i = 0; i < ephemTime.size(); i++) {
+    rotatedSunState = j2000_to_target.rotateStateAt(ephemTime[i], sunStates[i]);
+    sunPositions.push_back(rotatedSunState.position.x * 1000);

```

Comment:
maybe this is redundant, but I always appreciate comments like "ale stores in km, but usgscsm in meters" or whatever the case is here for the * 1000.